### ISSUE TITLE
Modal not capturing focus when autoFocus and keyboard are set to true

### ISSUE THREAD

**Issue by vanyadymousky** [reactstrap/reactstrap#874]
### Issue description

- components: Modal
- reactstrap version #5.0.0-beta
- import method es
- react version #16.2.0
- bootstrap version #4.0.0

### What is happening?

'Escape' key not working so modal cannot be closed via 'Escape'

### What should be happening?

Having 'autoFocus' and 'keyboard' props set to true, I should have an ability to close modal with 'Escape' button

### Code

```
<Modal
  isOpen={isOpen}
  fade={false}
  toggle={this.toggle}
  className={modalClassname}
>
  <ModalBody className={bodyClassName}>
    {children}
  </ModalBody>
</Modal>
```
### My workaround

Overriding onOpened method in modal, so I'm focusing manually (and it works!): 

```
export class ModalExtended extends Modal {
  onOpened() {
    super.onOpened();
    const { autoFocus, keyboard } = this.props;
    if (autoFocus && keyboard) {
      if (this._dialog && this._dialog.parentNode) {
        this._dialog.parentNode.focus();
      }
    }
  }
}
```

### Additional details

When debugging this, I've found that this._dialog is null in renderIntoSubtree (renderModalDialog is called after, so ref is set later than renderIntoSubtree is called): 

```
// check if modal should receive focus
      if (this._focus) {
        if (this._dialog // is null here. 
```

**Comment by dshitik** [comment]
+

**Comment by dshitik** [comment]
+

**Comment by dshitik** [comment]
++

**Comment by fantua** [comment]
Can you try reactstrap `5.0.0-beta.2` version?

**Comment by vanyadymousky** [comment]
@fantua Fixed with 5.0.0-beta.3, Thanks Igor!
